### ISSUE TITLE
Cannot read property 'kind' of undefined while ng build --prod

### ISSUE THREAD

**Issue by avishwakarma** [wynfred/ngx-masonry#43]
I am getting the following error while building my angular app for production.
Running ng serve does not produce any error

```
ERROR in /Users/ashok/WebRoot/sociostuff/node_modules/ngx-masonry/fesm5/ngx-masonry.js
Module build failed (from /Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
TypeError: Cannot read property 'kind' of undefined
    at isAngularDecoratorMetadataExpression (/Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:265:35)
    at checkNodeForDecorators (/Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:77:21)
    at visitNodes (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16144:30)
    at Object.forEachChild (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16370:24)
    at checkNodeForDecorators (/Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16135:24)
    at Object.forEachChild (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16265:21)
    at checkNodeForDecorators (/Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16135:24)
    at Object.forEachChild (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16322:24)
    at checkNodeForDecorators (/Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16135:24)
    at Object.forEachChild (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16333:24)
    at checkNodeForDecorators (/Users/ashok/WebRoot/sociostuff/node_modules/@angular-devkit/build-optimizer/src/transforms/scrub-file.js:68:31)
    at visitNode (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16135:24)
    at Object.forEachChild (/Users/ashok/WebRoot/sociostuff/node_modules/typescript/lib/typescript.js:16229:21)
```
Versions of packages from `package.json`

```
"@angular/animations": "~7.2.0",
"@angular/cdk": "^7.2.0",
"@angular/common": "~7.2.0",
"@angular/compiler": "~7.2.0",
"@angular/core": "~7.2.0",
"@angular/forms": "~7.2.0",
"@angular/http": "~7.2.0",
"@angular/platform-browser": "~7.2.0",
"@angular/platform-browser-dynamic": "~7.2.0",
"@angular/platform-server": "~7.2.0",
"@angular/router": "~7.2.0",
"ngx-masonry": "^9.1.1",
```

**Comment by avishwakarma** [comment]
I think its a version mismatch issue. Can anyone please specify which version of `ngx-masonry` should I use for Angular V 7.0.2

**Comment by wynfred** [comment]
Thanks for reporting. If you google the error there're plenty of related issues and fixes. I'm not sure which is the right fix but obviously it is not an issue in this library.

**Comment by avishwakarma** [comment]
@wynfred The issue seems due to version mismatch as I mentioned earlier.

Using V 1.1.4 fixes the build issue. But it starts a new issue of page flickering while items are being placed.

Tried disabling animation using options `animations: {}` and also tried to set `transitionDuration ` to `0s` but both are not working.

**Comment by wynfred** [comment]
`animations: {}` is only available on 9.X. transitionDuration should work and I'm not sure what is the flickering you mentioned. https://masonry.desandro.com/options.html may help.

**Comment by avishwakarma** [comment]
`transitionDuration: 0` fixes the flickering issue in V 1.1.4.

Thanks now I can build my app with `"buildOptimizer": true`

PS: You should add this to your documentation that latest version only supports Angular >= 8
For Lower angular version use V1.14

**Comment by wynfred** [comment]
Yes, 9.X is marked as beta in the readme, but theoretically it should work with older versions. If a lot of users stay on pre 8.X and have this issue, I could take some time to investigate.
